pyproject.toml

```diff
@@ -116,6 +116,16 @@ addopts = [
 filterwarnings = [
     'ignore:Support for Awkward Arrays is currently experimental',
     'ignore:Outer joins on awkward\.Arrays',
+    # TODO: replace both lines above with this one once we figured out how prevent ImportPathMismatchError
+    # 'ignore::anndata._warnings.ExperimentalFeatureWarning',
+]
+# When `--strict-warnings` is used, all warnings are treated as errors, except those:
+filterwarnings_when_strict = [
+    "default::anndata._warnings.ImplicitModificationWarning",
+    "default:Transforming to str index:UserWarning",
+    "default:(Observation|Variable) names are not unique. To make them unique:UserWarning",
+    "default::scipy.sparse.SparseEfficiencyWarning",
+    "default::dask.array.core.PerformanceWarning",
```

Comment:
Would be great to just add those to `filterwarnings` and not use a custom option, but due to some quirk in how pytest imports those object paths, that raises `ImportPathMismatchError` as described above.

So I rather have a bit more complexity by using an extra option than making the list of `default` filters as complex as they were before this PR: https://github.com/scverse/anndata/blob/00f39eba8e412639094c87a62ba935f9d94b6622/.azure-pipelines.yml#L70-L78